### PR TITLE
[SDN-VNET]  Protobuf changes

### PR DESCRIPTION
**Pull request by nijosmsft**
Main changes
- Added SubnetReference to IpConfiguration to support SDN VNET 
- Added ApiVersion to VirtualNetworkRequest
- Added LogicalNetwork reference and cidr to VirtualNetwork

### CODE REVIEW

File: rpc/common/moc_common_common.proto [hunk 1]
```diff
@@ -358,3 +358,8 @@ enum CloudInitDataSource {
 	NoCloud = 0;
 	Azure = 1;
 }
+
+message ApiVersion {
```
Comment (sbgms) [inline rpc/common/moc_common_common.proto:362]: Minor: Ideally it should be Major.Minor.Build.Revision.

**Review by sbgms — DISMISSED** [review]

File: rpc/common/moc_common_common.proto (hunk 1 above)
Comment (nijosmsft) [reply rpc/common/moc_common_common.proto:362]: agree, i didnt want to add fields to protobuf that we are not using not.

**Review by sbgms — APPROVED** [review]

**Review by andkimsdn — APPROVED** [review]

**Pull request merged**